auditbeat/module/file_integrity/eventreader_windows_test.go

```diff
@@ -0,0 +1,883 @@
+		subDir := filepath.Join(tempDir, "MySubDir")
+		runPowershell(t, fmt.Sprintf(`
+			Write-Host "-> Deleting directory: %s"
+			Remove-Item -Path '%s' -Force
+		`, subDir, subDir))
+	}
+	testFSNotifyReaderOperation(t, "delete-directory", setup, runTest)
+	testETWReaderOperation(t, "delete-directory", setup, runTest)
+}
+
+func testFSNotifyReaderOperation(
+	t *testing.T,
+	operation string,
+	setup func(t *testing.T, tempDir string),
+	runTest func(t *testing.T, tempDir string),
+) {
+	t.Run(fmt.Sprintf("fsnotify/%s", operation), func(t *testing.T) {
+		dir := t.TempDir()
+
+		if setup != nil {
+			t.Logf("Running setup for operation: %s", operation)
+			setup(t, dir)
+			time.Sleep(inactivityTimeout)
+		}
+
+		fsNotifyR, err := NewEventReader(Config{
+			Paths:     []string{dir},
+			Recursive: true,
+		}, logp.NewLogger("test"))
+		require.NoError(t, err)
+
+		done := make(chan struct{})
+
+		eventsFS, err := fsNotifyR.Start(done)
+		require.NoError(t, err)
+		time.Sleep(2 * time.Second)
+
+		var (
+			rcvdFSEvents []any
+			wg           sync.WaitGroup
+			started      sync.WaitGroup
+		)
+
+		consumerCount := 1
+
+		started.Add(consumerCount)
+		wg.Add(consumerCount)
+
+		go consumerFunc(t, done, eventsFS, &wg, &started, &rcvdFSEvents)
+
+		started.Wait()
+		runTest(t, dir)
+
+		t.Log("Waiting for events to be processed...")
+		wg.Wait()
+
+		close(done)
+		time.Sleep(2 * time.Second)
+		os.RemoveAll(dir)
+
+		writeSamples(t, rcvdFSEvents, fmt.Sprintf("%s_fsnotify.json", operation))
+	})
+}
+
+func testETWReaderOperation(
+	t *testing.T,
+	operation string,
+	setup func(t *testing.T, tempDir string),
+	runTest func(t *testing.T, tempDir string),
+) {
+	t.Run(fmt.Sprintf("etw/%s", operation), func(t *testing.T) {
+		dir := t.TempDir()
+
+		if setup != nil {
+			t.Logf("Running setup for operation: %s", operation)
+			setup(t, dir)
+			time.Sleep(inactivityTimeout)
+		}
+
+		etwR, err := newETWReader(Config{
+			Paths:         []string{dir},
+			Recursive:     true,
+			FlushInterval: 2 * time.Second,
+		}, logp.NewLogger("test"))
+		require.NoError(t, err)
+		if *saveETWEvents {
+			etwR.(*etwReader).etwEventsC = make(chan *etw.RenderedEtwEvent)
+		}
+
+		done := make(chan struct{})
+
+		eventsETW, err := etwR.Start(done)
+		require.NoError(t, err)
+		time.Sleep(2 * time.Second)
+
+		var (
+			rcvdETWEvents         []any
+			rcvdETWOriginalEvents []any
+			wg                    sync.WaitGroup
+			started               sync.WaitGroup
+		)
+
+		consumerCount := 1
+		if *saveETWEvents {
+			consumerCount++
+		}
+
+		started.Add(consumerCount)
+		wg.Add(consumerCount)
+
+		if *saveETWEvents {
+			go consumeOriginalFunc(t, done, etwR.(*etwReader).etwEventsC, &wg, &started, &rcvdETWOriginalEvents)
+		}
+		go consumerFunc(t, done, eventsETW, &wg, &started, &rcvdETWEvents)
+
+		started.Wait()
+		runTest(t, dir)
+
+		t.Log("Waiting for events to be processed...")
+		wg.Wait()
+
+		close(done)
+		time.Sleep(2 * time.Second)
```

Comment:
What is the sleep for?

We know that sleeps are source of brittleness in our tests, so I wonder if this can be written without them (possibly through the use of helpers like https://pkg.go.dev/testing/synctest (go1.25) or https://pkg.go.dev/github.com/stretchr/testify/assert#Eventually).